pkg/datapath/loader/compile.go

```diff
@@ -192,7 +194,26 @@ func compileAndLink(ctx context.Context, prog *progInfo, dir *directoryInfo, deb
 	return err
 }
 
-// progLDFlags determines the compiler flags for the specified prog and paths.
+var (
+	unameOnce sync.Once
+
+	// default fallback
+	machineName = "x86_64"
+)
+
+// getMachineName returns the machine hardware name of this host.
+func getMachineName() string {
+	unameOnce.Do(func() {
+		var uts unix.Utsname
+		err := unix.Uname(&uts)
+		if err == nil {
+			machineName = string(uts.Machine[:bytes.IndexByte(uts.Machine[:], 0)])
+		}
```

Comment:
Please at least log at debug level (or higher) on a failure here, with the error. Ignoring an error and falling back to x86-64 here means that this code could be completely broken and we'd never notice.

Reply:
Good point. I think this might even deserve a warning?